rootston/cursor.c

```diff
@@ -271,6 +271,12 @@ static void roots_cursor_press_button(struct roots_cursor *cursor,
 			break;
 		case WLR_BUTTON_PRESSED:
 			roots_seat_set_focus(seat, view);
+			if (surface && !view) {
+				struct wlr_layer_surface *layer = surface->role_data;
```

Comment:
This is very meh™. We want to get rid of `role_data`, and it's an internal field.

Reply:
We need some means of getting a wlr_layer_surface/wlr_xdg_surface/etc from a wlr_surface, be it this or some other way

Reply:
Just iterate over the layer surfaces list.

Reply:
That is Very Meh™.

Reply:
https://github.com/swaywm/wlroots/issues/527

Find another way. Set `wlr_surface.data` for instance.

Reply:
Meh.

Reply:
We need two functions:

`wlr_layer_surface_from_wlr_surface`

`wlr_surface_is_layer_surface`

Same for xwayland, xdg-shell, etc. This is out of scope for this PR, come back and fix this later.